test/util/tile_cover.cpp

```diff
@@ -0,0 +1,118 @@
+#include <mbgl/util/tile_cover.hpp>
+#include <mbgl/util/geo.hpp>
+#include <mbgl/map/tile_id.hpp>
+
+#include <gtest/gtest.h>
+
+#include <unordered_set>
+
+using namespace mbgl;
+using set = std::unordered_set<TileID>;
+
+TEST(TileCover, Empty) {
+    auto result = tileCover(LatLngBounds::empty(), 0, 0);
+    ASSERT_TRUE(result.empty());
+}
+
+TEST(TileCover, Arctic) {
+    auto result = tileCover(LatLngBounds::hull({ 86, -180 }, { 90, 180 }), 0, 0);
+    ASSERT_TRUE(result.empty());
+}
+
+TEST(TileCover, Antarctic) {
+    auto result = tileCover(LatLngBounds::hull({ -86, -180 }, { -90, 180 }), 0, 0);
+    ASSERT_TRUE(result.empty());
+}
+
+TEST(TileCover, WorldZ0) {
+    auto result = tileCover(LatLngBounds::world(), 0, 0);
+    ASSERT_EQ(1, result.size());
+    ASSERT_EQ(0, result[0].z);
+    ASSERT_EQ(0, result[0].x);
+    ASSERT_EQ(0, result[0].y);
+}
+
+TEST(TileCover, WorldZ1) {
+    auto result = tileCover(LatLngBounds::world(), 1, 1);
+    ASSERT_EQ(4, result.size());
+    ASSERT_EQ(
+        (set {{
+            TileID(1, 1, 1, 1),
+            TileID(1, 0, 1, 1),
+            TileID(1, 1, 0, 1),
+            TileID(1, 0, 0, 1)
+        }}),
+        (set {
+            result.begin(),
+            result.end()
+        }));
+}
+
+//TEST(TileCover, SingletonZ0) {
```

Comment:
`tileCover(LatLngBounds::singleton(...))` is always empty; IMO this is a bug in `tileCover` that should be fixed. It should return the containing tile.